Month3/LinkedList/LinkedList/TSLList.h

```diff
@@ -0,0 +1,334 @@
+#pragma once
+# include <iostream>
+#include <string>
+#include <functional>
+template <typename T>
+class TSLList
+{
+private:
+	
+	struct FNode
+	{
+		T Element;
+		FNode* Next = nullptr ;
+	};
+	
+	FNode* Head = nullptr;
+	FNode* Tail = nullptr;
+	
+	int Size = 0;
+public:
+	TSLList()
+	{
+
+	}
+	TSLList(const T& NewElement)
+	{
+		Head = new FNode();
+		Head->Element = NewElement;
+		Tail = Head;
+		Size = 1;
+	}
+
+	TSLList(const TSLList<T>& OtherList)
+	{
+		FNode* Current = OtherList.Head;
+		for (int i = 0; i < OtherList.Size; i++)
+		{
+			AddTail(Current->Element);
+			Current = Current->Next;
+		}
+	}
+
+	TSLList& operator=(const TSLList<T>& OtherList)
+	{
+		Clear();
+		FNode* Current = OtherList.Head;
+		for (int CurrentIndex = 0; CurrentIndex < OtherList.Size; CurrentIndex++)
+		{
+			AddTail(Current->Element);
+			Current = Current->Next;
+		}
+		return *this;
+	}
+
+	~TSLList()
+	{
+		Clear();
+	}
+
+
+
+	//Access
+	
+
+	T& operator[](const int Index) 
+	{
+		FNode* Current = Head;
+		for (int CurrentIndex = 0; CurrentIndex < Index; CurrentIndex++) 
+		{
+			Current = Current->Next;
+		}
+		
+		return Current->Next;
+	}
+	const T& operator[](const int Index) const
+	{
+		FNode* Current = Head;
+		for (int CurrentIndex = 0; CurrentIndex < Index; CurrentIndex++)
+		{
+			Current = Current->Next;
+		}
+
+		return Current->Next;
+	}
+
+
+	T& GetHead() 
+	{
+		return Head->Element;
+	}
+	const T& GetHead() const
+	{
+		return Head->Element;
+	}
+
+	T& GetTail() 
+	{
+		return Tail->Element;
+	}
+	const T& GetTail() const
+	{
+		return Tail->Element;
+	}
+	
+
+
+
+	//Size-------------------------------------
+
+	bool IsEmpty()
+	{
+		return Size == 0;
+	}
+
```

Comment:
remember to use separators between functions, coding standards 😉

Reply:
ok